examples/llm/src/mosaic_gpt.py

```diff
@@ -525,10 +526,26 @@ def loss(self, outputs, batch):
     def get_metrics(self, is_train=False):
         return self.train_metrics if is_train else self.eval_metrics
 
-    def update_metric(self, batch, outputs, metric):
-        outputs = outputs.view(-1, outputs.size(-1))
-        targets = self.get_targets(batch).view(-1)
-        metric.update(outputs, targets)
+    def update_metric(self, batch, outputs, metric) -> None:
+        if isinstance(metric, InContextLearningMetric):
+            if batch.get('mode', None) == 'icl_task':
+                # only apply ICL metrics to specially constructed
+                # icl_task batches
+                targets = self.get_targets(batch)
+                metric.update(batch, outputs, targets)
+        else:
+            outputs = outputs.view(-1, outputs.size(-1))
+            targets = self.get_targets(batch).view(-1)
+            metric.update(outputs, targets)
+
+    def add_eval_metrics(self, evaluator):
```

Comment:
Can we just put the ICL metrics on `MosaicGPT.eval_metrics` in the constructor? Instead of this extra function.

Reply:
Yes, should I also do the same if it's a `HuggingFaceModel`? Got to suppport both

Reply:
Hm yeah, for now you can just directly modify model.eval_metrics in the HF init function? Or we can chat in a bit in the office

Reply:
Punting this to a future PR